File: JavaScript/Algorithms/String/reverseString.js
/**
 * Reverses the given string
 *
 * @param {string} string
 * @returns {string}
 */
function reverseString(string) {
    var reverseString = '';
    var length = string.length;

    for (var i = length - 1;i > -1;i--) {
        reverseString += string.charAt(i);
    }

    return reverseString;
}

File: Java/Algorithms/String/ReverseString.java
/**
 * Class ReverseString
 */
public class ReverseString {
    /**
     * Reverses the given string
     *
     * @param str a given string that must be reversed
     * @return String
     */
    public String reverseString(String str) {
        String reverseString = "";
        int length = str.length();

        for (int i = length - 1;i > -1;i--) {
            reverseString += str.charAt(i);
        }

        return reverseString;
    }
}
